src/main/kotlin/io/billie/organisations/orders/resource/OrderResource.kt

```diff
@@ -0,0 +1,147 @@
+    @ApiResponses(
+            value = [
+                ApiResponse(
+                        responseCode = "200",
+                        description = "Returns order matching the id",
+                        content = [
+                            (Content(
+                                    mediaType = "application/json",
+                                    schema = Schema(implementation = OrderDto::class)
+                            ))]
+                ),
+                ApiResponse(responseCode = "400", description = "Bad request", content = [Content()])]
+    )
+    fun getById(@Valid @PathVariable orderId: UUID): OrderDto {
+        try {
+            val order = orderService.findById(OrderId(orderId))
+            return mapToDto(order)
+        } catch (e: OrderNotFoundException) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, e.message)
+        }
+    }
+
+    @PostMapping("/orders/{orderId}/shipments")
+    @ApiResponses(
+            value = [
+                ApiResponse(
+                        responseCode = "200",
+                        description = "Registered shipment dispatch for an order",
+                        content = [
+                            (Content(
+                                    mediaType = "application/json",
+                                    array = (ArraySchema(schema = Schema(implementation = Entity::class)))
+                            ))]
+                ),
+                ApiResponse(responseCode = "400", description = "Bad request", content = [Content()])]
+    )
+    fun post(@Valid @PathVariable orderId: UUID,
+             @Valid @RequestBody shipmentRequest: ShipmentRequest): ShipmentId {
+        try {
+            return orderService.notifyShipment(mapToDomain(orderId, shipmentRequest))
+        } catch (e: OrderNotFoundException) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, e.message)
+        } catch (e: IllegalArgumentException) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, e.message)
+        } catch (e: ShipmentAmountExceedOrderTotalException) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, e.message)
+        }
+    }
+
+    private fun mapToDtos(orders: List<Order>): List<OrderDto> {
+        return orders.stream()
+                .map(this::mapToDto)
+                .collect(Collectors.toList())
+    }
+
+    private fun mapToDto(order: Order): OrderDto {
+        return OrderDto(order.orderId!!.id, order.organisationId.id, order.orderAmount.amount)
+    }
+
+    private fun mapToDomain(organisationId: UUID, orderRequest: OrderRequest): Order {
+        return Order(null, Entity(organisationId), OrderAmount(orderRequest.amount), mutableSetOf())
+    }
+
+    private fun mapToDomain(orderId: UUID, shipmentRequest: ShipmentRequest): Shipment {
+        return Shipment(null, OrderId(orderId), ShipmentAmount(shipmentRequest.amount))
+    }
```

Comment:
I prefer to use separate domain and dto objects which allow control over what internal domain information is exposed outside domain and it also helps to transform or convert some information at times.